pylabrobot/resources/resource.py

```diff
@@ -147,6 +153,31 @@ def assign_child_resource(self, resource: Resource, location: Optional[Coordinat
 
     self.children.append(resource)
 
+  def check_assignment(self, resource: Resource, reassign: bool = True):
+    """Check if the resource assignment produces unsupported or dangerous conflicts.
+    """
+    msgs = []
+
+    # Check for self assignment
+    if resource is self:
+      msgs.append(f"Will not assign resource '{self.name}' to itself.")
+
+    # Check for reassignment to the same parent
+    if resource.parent is not None and (resource.parent is self):
+      msgs.append(f"Will not reassign resource '{resource.name}' " +
+                  f"to the same parent: '{resource.parent.name}'.")
```

Comment:
Perhaps it's better to make this a warning (using [the logger](https://github.com/PyLabRobot/pylabrobot/blob/86ff89651bc41de093a9840613ba7a78136b55fe/pylabrobot/__init__.py#L7))? Something like `f"Resource '{resource.name}' already assigned to '{resource.parent.name}'"`. I'm somewhat hesitant to raising errors when something is technically OK.

It would crash if `reassign` is False.

Reply:
Done :)

Note that now the downstream check (in `Deck._check_name_exists`)  is raised instead.

![image](https://user-images.githubusercontent.com/3259326/229383489-b3ccd25b-69bd-4447-9bef-6c5325d0c482.png)

The logger shows: 

`2023-04-02 19:53:24,904 - pylabrobot.resources.resource - WARNING - Resource 'plate_01' already assigned to 'basic deck 1'`

Reply:
Output with reassign=False:

![image](https://user-images.githubusercontent.com/3259326/229383581-a2608257-3ced-44b2-813e-e5ff179ab1b0.png)

Reply:
Great!